Sources/FunctionalTableData/TableSectionHeaderFooter.swift

```diff
@@ -27,6 +27,10 @@ public protocol TableHeaderFooterStateType: Equatable {
 }
 
 public struct TableSectionHeaderFooter<ViewType: UIView, Layout: TableItemLayout, S: TableHeaderFooterStateType>: TableHeaderFooterConfigType {
+	public func dequeueCell(from tableView: UITableView, at indexPath: IndexPath) -> UITableViewCell {
+		return UITableViewCell()
+	}
```

Comment:
Why is this needed for `TableHeaderFooterConfigType`? `AnyCellConfigType` conformance?
Is there any consequence to returning an empty cell here - are we ever expecting this to be called on this type?

Would it make sense to have an `AnyTableHeaderConfigType` that leans on `func dequeueHeaderFooter(from tableView: UITableView) -> UITableViewHeaderFooterView?` instead?

Reply:
DIffable data sources fails with a crash because of a default implementation of `dequeueCell()` that returned `UITableViewCell.cellIdentifier`. Deleting that is in the diff here as well. I suspect it has to do with default implementations in protocol extensions and inheritance chains, sometimes those aren't predictable.

From what I can tell, `TableHeaderFooterConfigType` is also a `CellConfigType`, so that dequeueCell is necessary. I can't think of a scenario where we call that when using the TableHeaderFooterConfigType as a TableHeaderFooterConfigType, but if someone chooses to also use it as a CellConfigType and doesn't do the necessary work in their implementation, that could spell trouble. That seems remote enough as a possibility to do this.